src/components/Canvas.tsx

```diff
@@ -0,0 +1,343 @@
+import './ColorOverview';
+
+import React, { useEffect, useRef, useState } from 'react';
+import { useSelector } from 'react-redux';
+
+import { getElf } from '../reducers/fileReducer';
+import { getResolution } from '../reducers/userInputReducer';
+import { colors } from './Colors';
+import SectionInfoView from './SectionInfoView';
+
+import '../../resources/css/canvas.scss';
+
+const Canvas = () => {
+    const canvasRef = useRef<HTMLCanvasElement>(null);
+    const [c, setContext] = useState<CanvasRenderingContext2D | null>(null); // Why could this be |null ?
```

Comment:
There are a lot of places in this file where "c" and its' properties could be null because it's not rendered yet, which is a common problem here. See my comment in the PR for details.